specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-05-01-preview/containerInstance.json

```diff
@@ -0,0 +1,2611 @@
+              "$ref": "#/definitions/EncryptionProperties"
+            },
+            "initContainers": {
+              "type": "array",
+              "description": "The init containers for a container group.",
+              "items": {
+                "$ref": "#/definitions/InitContainerDefinition"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "extensions": {
+              "type": "array",
+              "description": "extensions used by virtual kubelet",
+              "items": {
+                "$ref": "#/definitions/DeploymentExtensionSpec"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "confidentialComputeProperties": {
+              "description": "The properties for confidential container group",
+              "$ref": "#/definitions/ConfidentialComputeProperties"
+            },
+            "priority": {
+              "type": "string",
+              "description": "The priority of the container group.",
+              "enum": [
+                "Regular",
+                "Spot"
+              ],
+              "x-ms-enum": {
+                "name": "ContainerGroupPriority",
+                "modelAsString": true
+              }
+            },
+            "containerGroupProfile": {
+              "description": "The reference container group profile properties.",
+              "$ref": "#/definitions/ContainerGroupProfileReferenceDefinition"
+            },
+            "standbyPoolProfile": {
+              "description": "The reference standby pool profile properties.",
+              "$ref": "#/definitions/StandbyPoolProfileDefinition"
+            },
+            "isCreatedFromStandbyPool": {
+              "readOnly": true,
+              "type": "boolean",
+              "description": "The flag indicating whether the container group is created by standby pool."
+            }
+          },
+          "required": [
+            "containers",
+            "osType"
+          ]
+        }
+      }
+    },
+    "ContainerGroupProfileReferenceDefinition": {
+      "description": "The container group profile reference.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Th container group profile reference id."
+        },
+        "revision": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The container group profile reference revision."
+        }
+      }
+    },
+    "StandbyPoolProfileDefinition": {
+      "description": "The standby pool profile reference.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The standby pool profile reference id."
+        },
+        "failContainerGroupCreateOnReuseFailure": {
```

Comment:
>failContainerGroupCreateOnReuseFailure [](http://example.com/codeflow?start=9&length=38)

Same recommendation as to the previous Boolean added

Reply:
Similar to this above one, even this will always be either True or False, let me know if that is not preferred and I can make the changes accordingly.

Reply:
Same as above

Reply:
Same as above, as this feature has been already released, it is expected to behave in a certain way by the customers. The variables that I have defined as boolean like isCreatedFromStandbyPool or failContainerGroupCreateOnReuseFailure , are expected to have True or False as their corresponding values. So, the code cannot be modified to use enum variables instead.